manager/state/raft/raft.go

```diff
@@ -799,7 +830,37 @@ func (n *Node) ProcessRaftMessage(ctx context.Context, msg *api.ProcessRaftMessa
 		return nil, ErrMemberRemoved
 	}
 
-	n.cluster.ReportActive(msg.Message.From)
+	// Reject vote requests from unreachable peers
+	if msg.Message.Type == raftpb.MsgVote {
+		member := n.cluster.GetMember(msg.Message.From)
```

Comment:
Actually I don't think checking `Active` would work. If the member requesting the election is not the leader and the member that receives the request is not the leader either, the first member would be considered inactive (because it only communicates with the leader).